src/snowflake/snowpark/dataframe.py

```diff
@@ -198,15 +222,17 @@ def _disambiguate(
         # We use the session of the LHS DataFrame to report this telemetry
         lhs._session._conn._telemetry_client.send_alias_in_join_telemetry()
 
-    lhs_prefix = _generate_prefix("l")
-    rhs_prefix = _generate_prefix("r")
+    suffix_provided = lsuffix or rsuffix
```

Comment:
Though unlikely, we should handle the case about `lsuffix==rsuffix`. Probably we should raise an error if they are provided and equal (and also add a test for it)

Reply:
Added to raise ValueError